### ISSUE TITLE
findByIdAndUpdate behavior changed from 5.5.2 to 5.5.3

### ISSUE THREAD

**Issue by cal6** [Automattic/mongoose#7734]
There is a behavioural change between mongoose 5.5.2 and 5.5.3 - I want to know if this is intentional or a bug.

We used to rely on the following pattern:

```
let obj = myModel.findByIdAndUpdate('some_id', {foo: "bar"}, {new: true});
await doSomethingWithTheObject(obj);
return obj.set('processed', true).save();
```

Starting from 5.5.3, last row throws an error - .set is not a function. Also virtual field obj.id does not exist. It seems, that `findByIdAndUpdate` now returns lean object instead of mongoose object. At the same time, docs have not changed and it still should return Query object unless "lean" parameter is used (which is not the case currently).

Tried (and managed to reproduce) with  NodeJS version 8.15.1 and latest 10.15.3 with MongoDB version 4.0.8.

**Comment by caub** [comment]
yea, similar that my reported bug #7735

**Comment by vkarpov15** [comment]
Sorry for the trouble. @Fonger fixed this issue in 5.5.4.
